ext/openssl/ossl_x509store.c

```diff
@@ -329,7 +328,17 @@ ossl_x509store_set_trust(VALUE self, VALUE trust)
 static VALUE
 ossl_x509store_set_time(VALUE self, VALUE time)
 {
-    rb_iv_set(self, "@time", time);
+    X509_STORE *store;
+    X509_VERIFY_PARAM *param;
+
+    GetX509Store(self, store);
+    param = X509_VERIFY_PARAM_new();
```

Comment:
This might technically not be a public OpenSSL API anymore; I can't find it referenced in the public manpages (it's still public in LibreSSL, but not OpenSSL).

Instead of creating a new `X509_VERIFY_PARAM` you can probably use `X509_STORE_get0_param` to get the pre-existing `X509_VERIFY_PARAM` out of the `X509_STORE` object, which can then have its time set (and doesn't need to be re-inserted since `X509_STORE_get0_param` returns an internal pointer).

ref: https://www.openssl.org/docs/man3.3/man3/X509_STORE_set1_param.html

Reply:
(Using `X509_STORE_get0_param` should also have fewer allocation/set failure modes, AFAICT.)

Reply:
ah thanks for that, I didn't realize it returned an inner pointer!

Reply:
`X509_STORE_get0_param and X509_STORE_get0_objects were added in OpenSSL 1.1.0`